include/image.h

```diff
@@ -150,18 +202,25 @@ class ImageArray
     Data _images {};
 
 public:
-    ImageT& operator[](size_t i) {return *(_images[i]);}
-    const ImageT& operator[](size_t i) const {return *(_images[i]);}
+    ImageT& operator [] (size_t i) {return *(_images[i]);}
+    const ImageT& operator [] (size_t i) const {return *(_images[i]);}
 
-    typename Data::iterator begin() noexcept {return _images.begin();}
-    typename Data::const_iterator cbegin() const noexcept {return _images.begin();}
+    ImageT & at(size_t i=0) const {return (*_images.at(i));}
 
+    typename Data::iterator begin() noexcept {return _images.begin();}
```

Comment:
Might be easier to do
```
using iterator = typename Data::iterator;
```
in this class.  It's also nice to have an iterator type on anything that is iterable, so when `auto` doesn't work you don't have to resort to `decltype` stuff.

Reply:
I'm going to include this, but unfortunately it does not seem to obviate the need for an explicit (c)begin/(c)end to enable range-based for loops.